File: apple/Sources/ReactNativeWatchOS/Modifiers/PrivacySensitive.swift
import SwiftUI

/// SwiftUI `.privacySensitive(_:)`. Marks the subtree as private content so
/// the system can redact it when the device is in a privacy-sensitive state.
/// Defaults to `true`.
enum RNWPrivacySensitiveModifier {
    @MainActor
    static func register(into r: RNWModifierRegistry) {
        r.register("privacySensitive") { view, params, _ in
            AnyView(view.privacySensitive(params.bool("value") ?? true))
        }
    }
}

File: src/modifiers/privacySensitive.ts
import { createModifier } from './createModifier';

/// SwiftUI `.privacySensitive(_:)`. Marks the subtree as containing private,
/// user-sensitive content so the system can redact it when appropriate
/// (e.g. on an inactive/wrist-down watch face). Defaults to `true`.
export function privacySensitive(value: boolean = true) {
  return createModifier('privacySensitive', { value });
}
